File: src/main/java/burrow/furniture/exec/ExecCommand.java
package burrow.furniture.exec;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import org.springframework.lang.NonNull;
import picocli.CommandLine;

import java.io.IOException;

@CommandLine.Command(
    name = "exec",
    description = "Take the first argument as a command to execute."
)
@CommandType(ExecFurniture.COMMAND_TYPE)
public class ExecCommand extends Command {
    @CommandLine.Parameters(index = "0", description = "The command to execute.")
    private String command;

    public ExecCommand(@NonNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() throws IOException, InterruptedException {
        useFurniture(ExecFurniture.class).execute(commandContext, command);

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/carton/cradle/ExecCommand.kt
package burrow.carton.cradle

import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine
import picocli.CommandLine.Parameters

@CommandLine.Command(
    name = "exec",
    description = ["Executes a command."]
)
class ExecCommand(data: CommandData) : Command(data) {
    @Parameters(index = "0", description = ["Command to execute."])
    private var command = ""

    override fun call(): Int {
        val cradle = use(Cradle::class)
        return cradle.executeCommand(command, environment, stdout, stderr)
    }
}